test/python/topology/test2_namedtuple.py

```diff
@@ -151,58 +436,420 @@ def test_numbers_spl_py(self):
         st.for_each(check_numbers_tuple())
+        if debug_named_tuple_output:
+            st.print()
 
         tester = Tester(topo)
         tester.tuple_count(st, 3)
+        if debug_named_tuple_output:
+            self.test_config['topology.keepArtifacts'] = True
         tester.test(self.test_ctxtype, self.test_config)
 
-    # ValueError: Unsupported type: <class 'py36_types.SpottedSchema'>
-    def _test_list_of_tuple_spl_py(self):
-        # spl source -> python sink (NamedTupleListOfTupleSchema)
-        topo = Topology()
+
+    def test_unsupported_types_for_conversion_to_from_py(self):
+        tc = 'test_unsupported_types_for_conversion_to_from_py'
+        topo = Topology(tc)
+        b = op.Source(topo, "spl.utility::Beacon",
+            schema='tuple<set<list<int64>> sli64>',
+            params = {'period': 0.1, 'iterations':3})
+        b.sli64 = b.output('{[(int64) IterationCount()],[0l]}')
+        s = b.stream
+        print ('-A- EXPECT ERROR CDISP9164E for unsupported type: tuple<set<list<int64>>>')
+        tester = Tester(topo)
+        tester.tuple_count(s, 3) # causes test to fail: type is not supported for conversion to or from Python
+        res = tester.test(self.test_ctxtype, self.test_config, assert_on_fail=False)
+        assert(False == res) # expected result: test failed
+        print ('-A- unsupported type check: PASSED')
+
+        topo = Topology(tc)
+        b = op.Source(topo, "spl.utility::Beacon",
+            schema='tuple<set<tuple<rstring a, rstring b>> stuple>',
+            params = {'period': 0.1, 'iterations':3})
+        b.stuple = b.output('{{a="a", b="b"}}')
+        s = b.stream
+        print ('-B- EXPECT ERROR CDISP9164E for unsupported type: tuple<set<tuple<rstring a, rstring b>>>')
+        tester = Tester(topo)
+        tester.tuple_count(s, 3) # causes test to fail: type is not supported for conversion to or from Python
+        res = tester.test(self.test_ctxtype, self.test_config, assert_on_fail=False)
+        assert(False == res) # expected result: test failed
+        print ('-B- unsupported type check: PASSED')
+
+        topo = Topology(tc)
+        b = op.Source(topo, "spl.utility::Beacon",
+            schema=NamedTupleSetOfListofTupleSchema,
+            params = {'period': 0.1, 'iterations':3})
+        b.slt = b.output('{[{start_time=(float64)0.1, end_time=(float64)0.2, confidence=(float64)0.5}]}')
+        s = b.stream
+        print ('-C- EXPECT ERROR CDISP9164E for unsupported type: set<list<tuple<float64 start_time,float64 end_time,float64 confidence>>>')
+        tester = Tester(topo)
+        tester.tuple_count(s, 3) # causes test to fail: type is not supported for conversion to or from Python
+        res = tester.test(self.test_ctxtype, self.test_config, assert_on_fail=False)
+        assert(False == res) # expected result: test failed
+        print ('-C- unsupported type check: PASSED')
+
+    def _test_spl_source_list_of_tuple_named_tuple_py_sink(self):
```

Comment:
`_test_spl_source_list_of_tuple_named_tuple_py_sink(self):`

This test case is hidden, but should test a supported scenario.

Reply:
callable returns named tuple not dict, that is the difference
restrictions to named tuple objects needs to be mentioned